### DISCUSSION TITLE
mitmweb: Any details on how to clear the request queue, or select all requests? Keyboard shortcuts?

### DISCUSSION THREAD

**Discussion by erawhctim** [mitmproxy/mitmproxy#5396] [Q&A]
I'm having trouble with the web UI, specifically around managing the history/queue of requests. I can navigate up and down with the arrow keys, but the delete key doesn't actually delete the item from the history, ~and there doesn't seem to be a way to select all & delete~. What am I missing?

Is there a docs page or more details explaining how best to navigate in the web UI somewhere?

**EDIT:** I see now that there is a `File > Clear All` option, so that helps. I'm still unsure how to select an arbitrary subset of requests, though. Basically have to delete them one by one, or delete the entire thing 😕

**Comment by KevinSJ** [comment]
Hello, I was having the same question. Looks like this part needs some documentation or make it more obvious, but I did reverse engineered some keys by looking at the test file here -- https://github.com/mitmproxy/mitmproxy/blob/main/web/src/js/__tests__/ducks/ui/keyboardSpec.tsx

So 'z' is for clearing all the flows (same as clicking on the file and clear all). 'd' is for deleting one flow. You also get some simple vim-style navigation (j,k). But I don't see how to select a subset of flows and/or deleting them.

**Reply by erawhctim** [reply]
Nice find! I _knew_ this had to be somewhere (and I agree, this is an area of improvement that the docs could benefit from).

Digging through the docs in more detail, I think I found a list of the keyboard shortcuts for the CLI tool, and I'm assuming they also apply to the web as well? 
<img width="965" alt="Screen Shot 2022-06-09 at 11 49 27 AM" src="https://user-images.githubusercontent.com/6600919/172891872-e7656c28-49a7-46d2-b304-e3a0223faa88.png">

**Comment by mhils** [comment]
shortcuts are currently defined here: https://github.com/mitmproxy/mitmproxy/blob/main/web/src/js/ducks/ui/keyboard.tsx

They should pretty much match the mitmproxy ones. Multi-select and delete is definitely something we should add at some point. I'd of course be happy to accept a PR that makes pressing the delete key delete the currently selected flows. :)

**Reply by KevinSJ** [reply]
@mhils I created a PR #5408 to make pressing delete key to delete the currently selected flow. I actually planning to add multi-select as well. Maybe I should create a separate discussion to discuss the behavior for multi-select?

**Reply by mhils** [reply]
Awesome! Yes please. You'll find that some parts of the app state already support multiple selections, but the UI bits of that haven't been tackled yet. Please feel free to shoot questions my way. :)
